src/material/schematics/ng-generate/mdc-migration/rules/theming-styles.ts

```diff
@@ -46,12 +46,39 @@ export class ThemingStylesMigration extends Migration<ComponentMigrator[], Schem
       return m.styles.isLegacyMixin(this.namespace, atRule);
     });
     if (migrator) {
-      migrator.styles.replaceMixin(this.namespace, atRule);
-    } else if (atRule.params.includes('all-legacy-component-themes') && atRule.parent) {
-      // We use an arbitrary migrator because searching for the right one
-      // doesn't matter since the all-component-theme mixin affects all
-      // components and it only needs to be replaced once.
-      this.upgradeData[0]?.styles.replaceAllComponentThemeMixin(atRule);
+      const mixinChange = migrator.styles.getMixinChange(this.namespace, atRule);
+      if (mixinChange) {
+        replaceAtRuleWithMultiple(atRule, mixinChange.old, mixinChange.new);
+      }
+    } else if (atRule.params.includes('all-legacy-component-') && atRule.parent) {
+      replaceAllComponentsMixin(atRule);
+    } else if (atRule.params.includes('legacy-core') && atRule.parent) {
+      // When migrating some components, add mat.core but also leave
+      // mat.legacy-core mixin for the non MDC components. Leave a comment to
+      // tell users to delete the legacy mixin once all components are migrated.
+      if (this.upgradeData.length !== MIGRATORS.length) {
+        const legacyCoreComment =
+          'TODO: Remove the legacy-core mixin once all legacy components are migrated';
```

Comment:
we should make sure all of the TODOs added by our script our easily findable - I'm thinking something like `TODO(mdc-migration): ...`